src/server/scripts/Spells/spell_shaman.cpp

```diff
@@ -609,6 +609,11 @@ class spell_sha_fire_nova : public SpellScriptLoader
                     return SPELL_FAILED_CUSTOM_ERROR;
                 }
 
+                float distance = 30.0f;
+                Creature* totem = GetCaster()->GetMap()->GetCreature(GetCaster()->m_SummonSlot[1]);
```

Comment:
ugh, really ugly

get the caster -> get the map -> look for a creature -> get the summon for the magic number summon slot 1

go for: get the summon slot and null check it if necessary and cast it to a creature if necessary too